gravitee-apim-gateway/gravitee-apim-gateway-resource/src/test/java/io/gravitee/gateway/resource/internal/ResourceManagerImplTest.java

```diff
@@ -0,0 +1,227 @@
+            new ResourceManagerImpl(
+                false,
+                classLoader,
+                reactable,
+                resourcePluginManager,
+                resourceClassLoaderFactory,
+                resourceConfigurationFactory,
+                applicationContext
+            );
+    }
+
+    @Test
+    void shouldNotLoadAnyResourceWhenReactableDoesNotHaveResource() {
+        when(reactable.dependencies(Resource.class)).thenReturn(Collections.emptySet());
+        cut.initialize();
+
+        verifyNoInteractions(resourcePluginManager, resourceClassLoaderFactory, resourceConfigurationFactory, applicationContext);
+    }
+
+    @Test
+    void shouldThrowAnIllegalStateExceptionWhenPluginNotFoundInTheRegistry() {
+        final Resource resource = buildResource();
+        when(reactable.dependencies(Resource.class)).thenReturn(Set.of(resource));
+
+        assertThrows(IllegalStateException.class, () -> cut.initialize());
+
+        assertThat(cut.containsResource(resource.getName())).isFalse();
+        verify(resourcePluginManager).get(resource.getType());
+        verifyNoInteractions(resourceClassLoaderFactory, resourceConfigurationFactory, applicationContext);
+    }
+
+    @Test
+    void shouldIgnoreAndReturnNullWhenClassNotFoundOccurred() {
+        final Resource resource = buildResource();
+        final ResourcePlugin resourcePlugin = mock(ResourcePlugin.class);
+
+        // Set up an NPE.
+        when(resourcePlugin.resource()).thenReturn(Fake.class, null, Fake.class);
+        when(reactable.dependencies(Resource.class)).thenReturn(Set.of(resource));
+        when(resourcePluginManager.get(resource.getType())).thenReturn(resourcePlugin);
+
+        cut.initialize();
+        assertThat(cut.containsResource(resource.getName())).isFalse();
+    }
+
+    @Test
+    void shouldIgnoreAndReturnNullWhenFailingToRemoveClassLoaderAfterAnException() throws IOException {
+        final Resource resource = buildResource();
+        final ResourcePlugin resourcePlugin = mock(ResourcePlugin.class);
+
+        // Set up an NPE.
+        when(resourcePlugin.resource()).thenReturn(Fake.class, null, Fake.class);
+        when(reactable.dependencies(Resource.class)).thenReturn(Set.of(resource));
+        when(resourcePluginManager.get(resource.getType())).thenReturn(resourcePlugin);
+
+        final DefaultClassLoader spy = spy(classLoader);
+
+        doThrow(new IOException("Mock exception")).when(spy).removeClassLoader(anyString());
+
+        cut =
+            new ResourceManagerImpl(
+                false,
+                spy,
+                reactable,
+                resourcePluginManager,
+                resourceClassLoaderFactory,
+                resourceConfigurationFactory,
+                applicationContext
+            );
+
+        cut.initialize();
+        assertThat(cut.containsResource(resource.getName())).isFalse();
+    }
+
+    @Test
+    void shouldNotIgnoreWhenResourceIsDisabled() {
```

Comment:
Yeah, I discovered that the enabled flag wasn't used with v2 definition. I don't really know why and to avoid introducing any breaking changes I decided to not change the behavior for v2 while introducing the use of this flag for the v4 definition.

Reply:
I'm wondering if we should add a comment explaining that. Because it looks like a copy/error but it is a real expected behavior (at least for v3 engine) 🤔